plugins/catalog-backend-module-github/src/lib/defaultTransformers.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GroupEntity, UserEntity } from '@backstage/catalog-model';
+import { graphql } from '@octokit/graphql';
+import { getTeamMembers, Team, User } from './github';
+
+/**
+ * Generic Transformer
+ *
+ * @public
+ */
+export interface Transformer<InputType, OutputType, OptionsType> {
+  (item: InputType, options: OptionsType, ctx: TransformerContext):
+    | Promise<OutputType>
+    | OutputType;
+}
+
+export interface TransformerContext {
+  client: typeof graphql;
+  query: string;
+  org: string;
+}
+/**
+ * Options for UserTransformer
+ *
+ * @public
+ */
+export interface UserTransformerOptions {
+  namespace?: string;
+}
+
+/**
+ * Transformer for GitHub users to UserEntity
+ *
+ * @public
+ */
+export interface UserTransformer
+  extends Transformer<User, UserEntity, UserTransformerOptions> {
+  (item: User, options: UserTransformerOptions):
+    | Promise<UserEntity>
+    | UserEntity;
+}
+
+export interface TeamTransformerOptions {
+  namespace?: string;
+}
+
+export interface TeamTransformerResponse {
+  groups: GroupEntity;
+  groupMembers: string[];
+}
+
+export interface TeamTransformer
+  extends Transformer<Team, TeamTransformerResponse, TeamTransformerOptions> {
+  (item: Team, options: TeamTransformerOptions, ctx: TransformerContext):
+    | Promise<TeamTransformerResponse>
+    | TeamTransformerResponse;
+}
+
+/**
+ * Default transformer for GitHub users to UserEntity
+ *
+ * @public
+ */
+export const defaultUserTransformer: UserTransformer = (
+  item: User,
+  options: UserTransformerOptions,
+) => {
```

Comment:
if you go for the simplification above to not name the types - you don't actually have to name the types here either, you can do just `export const defaultUserTransformer: UserTransformer = (item, options) => ...`

Reply:
Agreed simplified. Thanks for the suggestion